source/opt/inline_pass.cpp

```diff
@@ -284,333 +670,60 @@ bool InlinePass::GenInlineCode(
     if (rid != 0) callee_result_ids.insert(rid);
   });
 
-  // If the caller is a loop header and the callee has multiple blocks, then the
-  // normal inlining logic will place the OpLoopMerge in the last of several
-  // blocks in the loop.  Instead, it should be placed at the end of the first
-  // block.  We'll wait to move the OpLoopMerge until the end of the regular
-  // inlining logic, and only if necessary.
-  bool caller_is_loop_header = false;
-  if (call_block_itr->GetLoopMergeInst()) {
-    caller_is_loop_header = true;
+  // Inline the entry block of the callee function.
+  bool prevInstWasReturn = false;
+  if (!InlineEntryBlock(&callee2caller, &new_blk_ptr, &returnLabelId,
+                        &prevInstWasReturn, calleeFn, callee_result_ids,
+                        returnVarId)) {
+    return false;
   }
 
-  bool callee_begins_with_structured_header =
-      (*(calleeFn->begin())).GetMergeInst() != nullptr;
-
-  // Clone and map callee code. Copy caller block code to beginning of
-  // first block and end of last block.
-  bool prevInstWasReturn = false;
-  uint32_t singleTripLoopHeaderId = 0;
-  uint32_t singleTripLoopContinueId = 0;
-  uint32_t returnLabelId = 0;
+  // Inline blocks of the callee function other than the entry block.
   bool multiBlocks = false;
```

Comment:
I think we can get rid of this as well.  All it is interested in is if inlining created more than 1 basic block.  This can happen in two ways: the callee has multiple basic block or has an early return.  However, these turn into the same thing: if there is an early return, then there must be more than one basic block.  All you need to check is if the callee has more than 1 basic block.

There is no need for the variable.  Would replace the use in the call to `MoveCallerInstsAfterFunctionCall` by checking the size of `new_blocks`.

Reply:
done